documentation/modules/metrics/con-operator-restart-events.adoc

```diff
@@ -0,0 +1,115 @@
+* `reportingComponent`: For Strimzi restart events, this will always be `strimzi.io/cluster-operator`.
+* `source`: An older version of the above field, this will also always be `strimzi.io/cluster-operator`.
+* `type`: This can be either `Warning` or `Normal`, for all current Strimzi restart events, it is currently always `Normal`.
+
+==== Kubernetes 1.25+
+
+When Kubernetes 1.25 is released and installed on your cluster, you will also be able to use these fields with  `--field-selector`:
+
+* `regarding.kind`: same as `involvedObject.kind`
+* `regarding.namespace`: same as `involvedObject.namespace`
+* `regarding.name`: same as `involvedObject.name`
+* `regarding.name`: same as `involvedObject.uid`
+* `reportingController`: same as `reportingComponent`
+
+== Current reasons why Strimzi will restart a Kafka pod
+
+* CaCertHasOldGeneration - The pod is still using a server certificate signed with an old CA, so needs to be restarted as part of updating that certificate
+* CaCertRemoved - Expired CA certificates have been removed, and the pod is restarted to run with the current certificates
+* CaCertRenewed - CA certificates have been renewed, and the pod is restarted to run with these updated certificates
+* ClientCaCertKeyReplaced - The key used to sign client CA certificates has been replaced, and the pod is being restarted as part of the CA renewal process
+* ClusterCaCertKeyReplaced - The key used to sign the cluster's CA certificates has been replaced, and the pod is being restarted as part of the CA renewal process
+* ConfigChangeRequiresRestart - Some Kafka configuration properties can be changed dynamically, but others require that the broker be restarted
+* CustomListenerCaCertChanged - CA certificates used to secure the Kafka network listeners has changed, and the pod is restarted to use it
+* FileSystemResizeNeeded - The file system size has been increased, and a restart is needed to apply it
+* JbodVolumesChanged - A disk was added or removed from the Kafka volumes, and a restart is needed to apply the change
+* ManualRollingUpdate - A user annotated the pod, or the stateful set / strimzi pod set it belongs to, to trigger a restart
+* PodForceRestartOnError - An error occurred that requires restarting the pod to rectify
+* PodHasOldGeneration - The stateful set that the pod is a member of has been updated, so the pod needs to be recreated
+* PodHasOldRevision - The strimzi pod set that the pod is a member of has been updated, so the pod needs to be recreated
+* PodStuck  - The pod is still pending, and either unschedulable, or not scheduled, so the operator is going to restart it as a last ditch effort.
+* PodUnresponsive - Strimzi was unable to connect to the pod, which can indicate a broker not coming up properly, so the operator will restart it to try to resolve the issue
+* ServerCertUpdated - One or more TLS certificates used by the Kafka broker have been updated, and a restart is needed to use them
```

Comment:
Are all these changes still captured even with StrimziPodSets? As I suspect many of them will apply only to StatefulSets and StrimziPodSets will in most cases log `PodHasOldRevision`. If that is the case, it might be good to note that in the docs to make sure people do not expect these events anymore.

Reply:
I only know of one that's stateful set specific, but if you can think of any others that might be, I'll double check. :)